app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/LoadingActivity.java

```diff
@@ -1,98 +0,0 @@
-package edu.drexel.lapcounter.lapcounter.frontend;
-
-import android.arch.lifecycle.ViewModelProviders;
-import android.content.Intent;
-import android.support.v7.app.AppCompatActivity;
-import android.os.Bundle;
-import android.util.Log;
-import android.view.View;
-
-import java.util.ArrayList;
-import java.util.List;
-
-import edu.drexel.lapcounter.lapcounter.R;
-import edu.drexel.lapcounter.lapcounter.backend.Database.Device.DeviceRepository;
-import edu.drexel.lapcounter.lapcounter.backend.Database.State.State;
-import edu.drexel.lapcounter.lapcounter.backend.Database.Transition.TransitionRepository;
-import edu.drexel.lapcounter.lapcounter.backend.Database.Units.Units;
-import edu.drexel.lapcounter.lapcounter.backend.Database.Workout.Workout;
-import edu.drexel.lapcounter.lapcounter.backend.Database.Workout.WorkoutRepository;
-import edu.drexel.lapcounter.lapcounter.backend.Database.Workout.WorkoutViewModel;
-import edu.drexel.lapcounter.lapcounter.backend.TimestampConverter;
-
-public class LoadingActivity extends AppCompatActivity {
-
-    private final static String TAG = LoadingActivity.class.getSimpleName();
-
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_loading_screen);
-
-        // In the final version, use R.string.<string id> for titles
-        getSupportActionBar().setTitle("Splash Screen");
-
-
-        // Initialize database as needed.
-        initDatabase();
-    }
-
-    /**
-     * Call this when the splash screen is done
-     * to traansition to the Welcome screen.
-     *
-     * Right now, this is aattached to a button click, but
-     * really it should be on a timer.
-     */
-    public void finishLoading(View view) {
-        Intent intent = new Intent(this, WelcomeActivity.class);
-        startActivity(intent);
-    }
-
-    private void initDatabase() {
-        // Initialize the enum tables. ====================================
-        try
```

Comment:
What is in this try/catch is needed as these are tables that have to be populated in order to save workouts/transitions.  If you just add this function (minus the workout part) into the splash screen, that should fix our issue.

Reply:
I copied this code into the `SplashActivity`